tests/integration/simulacron/test_connection.py

```diff
@@ -114,3 +114,36 @@ def test_heart_beat_timeout(self):
 
         # In this case HostConnection._replace shouldn't be called
         self.assertNotIn("_replace", executor.called_functions)
+
+    def test_callbacks_and_pool_when_oto(self):
+
+        def callback(result, count):
+            next(count)
+
+        def errback(exc, count):
+            next(count)
+
+        start_and_prime_singledc()
+
+        cluster = Cluster(protocol_version=PROTOCOL_VERSION, compression=False)
+        session = cluster.connect()
+
+        self.addCleanup(cluster.shutdown)
+        self.addCleanup(stop_simulacron)
+
+        query_to_prime = "SELECT * from testkesypace.testtable"
+
+        prime_query(query_to_prime, then={"delay_in_ms": 2000})
+
+
+        count = itertools.count()
+        future = session.execute_async(query_to_prime, timeout=1)
+        future.add_callbacks(callback, errback, callback_args=(count, ), errback_args=(count, ))
+        self.assertRaises(OperationTimedOut, future.result)
+
+        assert_quiescent_pool_state(self, cluster)
```

Comment:
This fail before the fix because not all the request ID used to make where the OTO happens it's not returned.